server/neptune/gateway/event/source/credentials.go

```diff
@@ -0,0 +1,198 @@
+package source
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/bradleyfalzon/ghinstallation"
+	"github.com/google/go-github/v45/github"
+	"github.com/pkg/errors"
+)
+
+// GithubCredentials handles creating http.Clients that authenticate.
+type GithubCredentials interface {
+	Client() (*http.Client, error)
+	GetToken() (string, error)
+	GetUser() (string, error)
+}
```

Comment:
Why do we need this interface? and all the other non-github app implementations?

Can't we just plumb the installation token through and just use that in the gitconfig for cloning? Would eliminate all this legacy stuff.